pkg/util/tracingservice/client_test.go

```diff
@@ -0,0 +1,148 @@
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tracingservice
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/migration/nodelivenesstest"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/rpc/nodedialer"
+	"github.com/cockroachdb/cockroach/pkg/util"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/netutil"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
+	"github.com/cockroachdb/cockroach/pkg/util/tracingservice/tracingservicepb"
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func setUpService(
+	t testing.TB,
+	rpcContext *rpc.Context,
+	localNodeID roachpb.NodeID,
+	remoteNodeID roachpb.NodeID,
+	localTracer *tracing.Tracer,
+	remoteTracer *tracing.Tracer,
+) *TraceClientDialer {
+	s := rpc.NewServer(rpcContext)
+	remoteTraceServer := NewTraceService(remoteTracer)
+	tracingservicepb.RegisterTraceServer(s, remoteTraceServer)
+	ln, err := netutil.ListenAndServeGRPC(rpcContext.Stopper, s, util.TestAddr)
+	require.NoError(t, err)
+
+	s2 := rpc.NewServer(rpcContext)
+	localBlobServer := NewTraceService(localTracer)
+	require.NoError(t, err)
+	tracingservicepb.RegisterTraceServer(s2, localBlobServer)
+	ln2, err := netutil.ListenAndServeGRPC(rpcContext.Stopper, s2, util.TestAddr)
+	require.NoError(t, err)
+
+	localDialer := nodedialer.New(rpcContext,
+		func(nodeID roachpb.NodeID) (net.Addr, error) {
+			if nodeID == remoteNodeID {
+				return ln.Addr(), nil
+			} else if nodeID == localNodeID {
+				return ln2.Addr(), nil
+			}
+			return nil, errors.Errorf("node %d not found", nodeID)
+		},
+	)
+
+	testNodeLiveness := nodelivenesstest.New(2 /* numNodes */)
+	return NewTraceClientDialer(
+		localNodeID,
+		localDialer,
+		testNodeLiveness,
+		localTracer,
+	)
+}
+
+func TestTraceClientGetSpanRecordings(t *testing.T) {
+	localNodeID := roachpb.NodeID(1)
+	remoteNodeID := roachpb.NodeID(2)
+	localTracer := tracing.NewTracer()
+	remoteTracer := tracing.NewTracer()
+
+	stopper := stop.NewStopper()
+	clock := hlc.NewClock(hlc.UnixNano, time.Nanosecond)
+	rpcContext := rpc.NewInsecureTestingContext(clock, stopper)
+	rpcContext.TestingAllowNamedRPCToAnonymousServer = true
+
+	traceDialer := setUpService(t, rpcContext, localNodeID, remoteNodeID, localTracer, remoteTracer)
+	localTID, remoteTID := setupTraces(localTracer, remoteTracer)
+
+	ctx := context.Background()
+	t.Run("fetch-localTID-recordings", func(t *testing.T) {
```

Comment:
Here and below, drop TID from the sub-test name.

Reply:
done.